airbyte-integrations/connectors/source-twilio-taskrouter/source_twilio_taskrouter/twilio_taskrouter.yaml

```diff
@@ -0,0 +1,36 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://taskrouter.twilio.com"
+    http_method: "GET"
+    authenticator:
+      type: BasicHttpAuthenticator
+      username: "{{ config['account_sid'] }}"
+      password: "{{ config['auth_token'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  workspaces_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "workspaces"
+      primary_key: "sid"
+      path: "/v1/Workspaces/{{config['sid']}}"
```

Comment:
looks like this endpoint only returns one workspace, can you implement the one that list workspaces? I created an account, I am able to call the API without an `sid`. Though only one workspace is returned on my account. 

Besides, according to [the airbyte doc](https://docs.airbyte.com/connector-development/tutorials/cdk-tutorial-python-http/read-data/), it's expected to return a list of entities and each matches the schema. 

```
    def parse_response(
            self,
            response: requests.Response,
            stream_state: Mapping[str, Any],
            stream_slice: Mapping[str, Any] = None,
            next_page_token: Mapping[str, Any] = None,
    ) -> Iterable[Mapping]
```

Reply:
The doc points to the python cdk, however I have made this connector using the low code one. Does the concept apply the same? The issue i had with listing all the workspaces was that the json out put consisted of two objects meta and workspaces with further properties nested so i was unable to figure out how to assign a primary key in the yaml file

Reply:
the concept is the same, the low-code CDK is also called configuration based CDK. You can also implement a custom class and use it in this yaml file. 

Regarding to other streams, I could think of 2 routes, if you decided to make workspaceId* required, you can use it to build other streams easily. 

If you plan to make the workspaceId optional, you may firstly fetch workspaces first and cache the Ids in memory, and fetch other resources on EACH id. you need something in the following:
https://docs.airbyte.com/connector-development/cdk-python/http-streams/#nested-streams--caching

* workspaceId refers to sid in your spec

Reply:
> the json out put consisted of two objects meta and workspaces

Regarding the output parsing, you can check the method called parse_response in python cdk. In the low-code CDK, it refers to [Record Selector](https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/record-selector#selecting-the-whole-response)

Reply:
Regarding to other streams, I could think of 2 routes, if you decided to make workspaceId* required, you can use it to build other streams easily. You just reference the id by something like "{config.get('workspaceId')}/activities"

I think this method might be the most suitable as it would let us add the streams without much hassle.

Reply:
That works, it's okay to accept the limitation in the first version. If people want to fetch all resources under all workspaces, we can extend it later.

Reply:
Is it possible to give individual field pointers to different streams? Ive implemented the listing of all workspaces stream by giving the field pointer in the selector ["workspaces"]. However this causes other streams to be affected for example the single workspace stream is unable to display any output

Reply:
> Is it possible to give individual field pointers to different streams? 
My understanding to your question is that, "is it possible to share the cursor value to multiple streams?" Yes, you can, that's called state slice, and that's also why you need to set `use_cache` true. Each stream that requires a workspaceId in the URL path depends on the stream workspaces. It will fetch the workspaces and store them in memory. Since you only need the id, not the whole response, we call it slice. 

And it's also available in low-code CDK. 
https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/stream-slicers#list-stream-slicer

Reply:
Okay got it, I'll try to implement it and see how it goes..

Reply:
What will the schema look like if I have an array in the json output. For example it goes as
{  workspace: 
 [  
    { prop1, prop2, prop3}, {prop4, prop5, prop6 } 
 ]
}

Reply:
I got it working! I also got another stream workers running smoothly. Now all that's left is the paginator